TypeScript definitions for fuzzyset library

lgrignon opened the pull request:
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

andy-ms commented on types/fuzzyset/index.d.ts:
```diff
@@ -0,0 +1,14 @@
+// Type definitions for fuzzset 1.0
+// Project: https://github.com/axiak/fuzzyset
```
That's a python library!
The `fuzzyset` on NPM claims to have a homepage at http://github.com/washt/fuzzyset .
But that page claims to be published as `fuzzyset.js`, which is also a real NPM package.
So, what exactly are these types referring to? :confounded:

lgrignon replied:
My bad, it's https://github.com/Glench/fuzzyset.js and I fixed it for this PR

andy-ms replied:
That page claims to be published as `fuzzyset.js`, which would mean the directory here should be renamed. But that package hasn't been published since 2012.

andy-ms commented on types/fuzzyset/index.d.ts:
```diff
@@ -0,0 +1,14 @@
+    values(): string[];
+}
+
+declare function FuzzySet(source: string[], useLevenshtein?: boolean, gramSizeLower?: number, gramSizeUpper?: number): FuzzySet;
```
Note that the one on NPM at least seems to be available as a module instead of as a global. So it should have `export = FuzzySet;` and be imported with `import FuzzySet = require("fuzzyset");` if so.

lgrignon replied:
Yes you are right, that's the syntax I use in my project, I recently added the module way so I join it to this PR (and add import in tests as well).
Still, the global way is also possible, both are coexistent in this library.

lgrignon commented:
I apologize but dtslint seems broken since neither ../tslint.json nor dtslint/dt.json works.

andy-ms commented:
Be sure to install the latest version of dtslint. `dtslint.dt.json` is the correct "extends" value.
The linter will complain about the `declare module`. Our preferred method to declare both a module and a global is:

```ts
declare function FuzzySet(...);
export = FuzzySet;
export as namespace FuzzySet;
```

lgrignon commented:
ok nearly as many commits as lines in the definition file, but we should be close to the end :)
Thanks for the export syntax, this is what I was looking for. I renamed to fuzzyset.js, then switched back to fuzzyset : I finally got what you meant. Actually there are 3 fuzzyset libraries (fuzzy, isn't it?), and `fuzzyset` is a fork of `fuzzyset.js`, so finally, I changed the target library to be `fuzzyset`.
Anyway, thanks for your patience, I hope it is ok now.

andy-ms commented:
Thanks!
